### PR TITLE
factoring out postgres creation

### PR DESCRIPTION
**Pull request by paulharter**
Big fat PR with slightly too many things:

- Removes the Postgres generation code which now lives in the console
- Simplifies file creation by moving all metadata and generated satellite code into the manifest
- Builds index.js bundle for distribution directly from servers version of migrations
- Uses `app_id` and `environment` in api
- Simplified CLI commands
- Updated docs
- Factor out `Migration` struct 
- sha256 now generated from normalised version of original sql code without comments

### CODE REVIEW

**Comment by paulharter** [comment]
refactoring to moving the postgres creation into a separate module in preparation for doing it on the server instead

File: lib/electric/migrations/templates/migration.sql.eex [hunk 1]
```diff
@@ -1,4 +1,10 @@
 /*
-ElectricDB Migration
-{"metadata": {"title": "<%= title %>", "name": "<%= name %>"}}
+Electric SQL Migration
+name: <%= name %>
+title": <%= title %>
```
Comment (icehaunter) [inline lib/electric/migrations/templates/migration.sql.eex:4]: Possibly extra `"` after the title?
I wonder, why did you decide to make this data unparsable by the machine? Or is this a switch to YAML in a way?

File: lib/electric/migrations/migrations.ex [hunk 2]
```diff
@@ -19,7 +19,7 @@ defmodule Electric.Migrations do
   optional argument:
   - :dir where to create the migration rather than the current working directory
   """
-  def init_migrations(options) do
+  def init_migrations(app_id, options) do
```
Comment (icehaunter) [inline lib/electric/migrations/migrations.ex:22]: What about the concept of enviornments, or is this a composite id (e.g. `default.lorem-ipsum-245`?)

File: lib/electric/migrations/sync.ex [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        "apps/#{app_id}/environment/#{environment}/migrations?body=satellite"
+      else
+        "apps/#{app_id}/environment/#{environment}/migrations"
+      end
+
+    case Client.get(url) do
+      {:ok, %Req.Response{status: 200, body: data}} ->
+        {:ok, Jason.decode!(data)}
+
+      {:ok, _stuff} ->
```
Comment (icehaunter) [inline lib/electric/migrations/sync.ex:29]: `IO.inspect/1` probably shouldn't stay here in published code, right?

File: lib/electric/migrations/sync.ex [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        {:error, "failed to connect"}
+    end
+  end
+
+  def get_full_migration_from_server(app_id, environment, migration_name) do
+    url = "apps/#{app_id}/environment/#{environment}/migrations/#{migration_name}?body=all"
+
+    case Client.get(url) do
+      {:ok, %Req.Response{status: 200, body: data}} ->
+        {:ok, Jason.decode!(data)}
```
Comment (icehaunter) [inline lib/electric/migrations/sync.ex:42]: I'm about 90% sure this call is not needed and will fail in production, as `req` does automatic body decoding

Comment (icehaunter) [reply lib/electric/migrations/sync.ex:42]: If it's required in development, then you probably forgot to set `Content-Type: application/json` header somewhere in the mock server

**Review by icehaunter — CHANGES_REQUESTED** [review]

File: lib/electric/migrations/templates/migration.sql.eex (hunk 1 above)
Comment (paulharter) [reply lib/electric/migrations/templates/migration.sql.eex:4]: As this is all in the manifest now I didnt want a machine readable version in two palces

File: lib/electric/migrations/migrations.ex (hunk 2 above)
Comment (paulharter) [reply lib/electric/migrations/migrations.ex:22]: The local migration folder covers all the environments - I thought capturing and storing the app_id here would make the rest of CLI commands neater as its unlikly to change. It's just lorem-ipsum-245 not a composite one

File: lib/electric/migrations/sync.ex (hunk 3 above)
Comment (paulharter) [reply lib/electric/migrations/sync.ex:29]: thnaks

File: lib/electric/migrations/sync.ex (hunk 4 above)
Comment (paulharter) [reply lib/electric/migrations/sync.ex:42]: ok Ill check the mock and updater this

**Pull request merged**